### ISSUE TITLE
Mass calibration

### ISSUE THREAD

**Issue by tomas-pluskal** [mzmine/mzmine3#93]
I would like to add a module for m/z calibration of peak lists based on compounds with known chemical formulas.

Something like this:

![screen shot 2016-02-16 at 13 54 03](https://cloud.githubusercontent.com/assets/1796692/13087212/c990f832-d4b4-11e5-9d6e-f69a07a6646f.png)

**Comment by dyrlund** [comment]
Good idea. What about also allowing to perform a RT calibration? This could for example be used to correct RT values based on internal standards before performing identification using a custom database.

And we could also consider quantile or bin calibration in addition to a full range calibration.

**Comment by photocyte** [comment]
I'd also add that it would be nice in all cases of calibration the "uncorrected" value (for RT m/z or whatever else), is saved within the data structure.  My ideal case would be to have the data structure support the regeneration of a "workably identical" mzML file, including metadata, from the internal data representation, as well as a corrected mzML file containing the altered values if desired.  Such manipulations should be clearly recorded in the mzML however.

**Comment by dyrlund** [comment]
This is already supported in the updated data structure in MSDK. So when you for example normalize the areas of features you will get a new column with the normalized areas while also preserving the original areas. In MZmine 2 only the normalized areas were stored.

With regards to the m/z and RT we probably need to allow some modules (like the identification modules) to either work on the original data or the corrected data (e.g. calibrated) if this is available.
